pkg/skaffold/kubernetes/owner.go

```diff
@@ -19,50 +19,93 @@ package kubernetes
 import (
 	"fmt"
 
-	"github.com/sirupsen/logrus"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
 )
 
 // TopLevelOwnerKey returns a key associated with the top level
 // owner of a Kubernetes resource in the form Kind-Name
-func TopLevelOwnerKey(obj metav1.Object, kind string) string {
-	for {
-		or := obj.GetOwnerReferences()
-		if or == nil {
-			return fmt.Sprintf("%s-%s", kind, obj.GetName())
-		}
-		var err error
-		kind = or[0].Kind
-		obj, err = ownerMetaObject(obj.GetNamespace(), or[0])
-		if err != nil {
-			logrus.Warnf("unable to get owner from reference: %v", or[0])
-			return ""
-		}
+func TopLevelOwnerKey(obj metav1.Object, kind string) (string, error) {
+	client, err := Client()
+	if err != nil {
+		return "", err
+	}
+	discovery := client.Discovery()
+
+	dynClient, err := DynamicClient()
+	if err != nil {
+		return "", err
+	}
+
+	child := objectWrapper{
+		Object: obj,
+		kind:   kind,
 	}
+
+	topLevel, err := topLevel(child, discovery, dynClient)
+	if err != nil {
+		return "", err
+	}
+
+	id := fmt.Sprintf("%s-%s", topLevel.GetKind(), topLevel.GetName())
```

Comment:
is this `Kind` and `Name` enough to uniquely identify a object. I am thinking a case where there are workloads across namespace.

Reply:
this is a copy of what was there before. It's the part of a larger key that identifies one port forwarding entry

Reply:
Each object is uniquely identified by its namespace, kind, and metadata.name ([source](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/)).  We should include the namespace here.